### DISCUSSION TITLE
Unable to adjust volume on my display when an external speaker is connected

### DISCUSSION THREAD

**Discussion by cansurmeli** [MonitorControl/MonitorControl#628] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] I have read through [the Monitor-Troubleshooting Wiki](https://github.com/MonitorControl/MonitorControl/wiki/Monitor-Troubleshooting)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the issue

My Mac is connected to an external screen where the said external screen is also connected to my external speakers.

I've searched through existing issues along with the Wiki and `ddcctl` seems **unable** to control the volume as well when I invoke it from the Terminal.

As the Wiki mentions, it probably won't work but I am opening this issue to ask is there anything else I can do or is there an update planned that could solve this issue?

Thanks.

### Expected behavior

Upon using the volume keys on my keyboard, the volume of the Mac should change accordingly.

### Environment Information (please complete the following information)

```markdown
- macOS version: 11.6 (20G165)
- Mac model: MacBook Pro (15-inch, 2017)
- MonitorControl version: 3.1.1 Build 3096 - Intel
- Monitor(s): Dell U2520D
- Monitor Cable(s)/Connection(s): over USB-C
- Apple Silicon/M1 (yes or no): no
```

**Comment by waydabber** [comment]
If DDC control works otherwise (display backlight is changing for example) only volume can't be controlled then it is probably a display issue. There were reports of several displays that have this issue. You should experiment with ddcctl and see if DDC options like `speakerSelect` makes any difference as it might make the display to switch control to the external display. Also `audioJackConnectionStatus` might be interesting. If you find that your display needs a speakerSelect to work, we can incorporate such a feature into MonitorControl (note: up until now nobody reported that such commands fix these kinds of problems so currently there is no option to change the speakerSelect option for volume).

Here are the audio related control codes you can experiment with (it is rare that display manufacturers implement most or any of these commands besides `audioSpeakerVolume`, but every display is different.

```
  case audioSpeakerVolume = 0x62
  case speakerSelect = 0x63
  case audioMicrophoneVolume = 0x64
  case audioJackConnectionStatus = 0x65
  case audioMuteScreenBlank = 0x8D
  case audioTreble = 0x8F
  case audioBass = 0x91
  case audioBalanceLR = 0x93
  case audioProcessorMode = 0x94
```

**Comment by waydabber** [comment]
Please let me know if you have any results regarding this. I'll convert this issue to a discussion as this issue comes up from time to time.
